java/tests/com/google/privacy/differentialprivacy/CountTest.java

```diff
@@ -512,4 +512,156 @@ private static void testForBias(
+  }
+
+  @Test
+  public void computeConfidenceInterval_gaussianTest() {
+    // Mock the noise mechanism.
+    when(noise.computeConfidenceInterval(anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new GaussianNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0],  (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3],  (Double) invocation.getArguments()[4],  (Double) invocation.getArguments()[5]));
+    count =
+            Count.builder()
+                    .epsilon(0.5)
+                    .delta(0.9)
+                    .noise(noise)
+                    .maxPartitionsContributed(15)
+                    .build();
+    count.increment();
+    count.computeResult();
+
+    // The result interval = (-1, 3), but count can't be equal to 0, therefore it will be clamped to 0.
+    assertThat(count.computeConfidenceInterval(0.152145599)).isEqualTo(ConfidenceInterval.create(0,3));
+  }
+
+  @Test
+  public void computeConfidenceInterval_laplaceTest() {
+    // Mock the noise mechanism. Since noise is not Laplace, nor Gaussian, delta will be passed as a value instead of null, in order to pass the checks.
+    when(noise.computeConfidenceInterval(anyLong(), anyInt(), anyLong(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new LaplaceNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0],  (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3],  null,  (Double) invocation.getArguments()[5]));
+    count =
+            Count.builder()
+                    .epsilon(0.1)
+                    .noise(noise)
+                    .delta(0.5)
+                    .maxPartitionsContributed(1)
+                    .build();
+    count.incrementBy(10);
+    count.computeResult();
+
+    assertThat(count.computeConfidenceInterval(0.5)).isEqualTo(ConfidenceInterval.create(3,17));
+  }
+
+  @Test
+  public void computeConfidenceInterval_whenComputeResultWasNotCalled_throwsException() {
+    // Mock the noise mechanism. Since noise is not Laplace, nor Gaussian, delta will be passed as a value instead of null, in order to pass the checks.
+    when(noise.computeConfidenceInterval(anyDouble(), anyInt(), anyDouble(), anyDouble(), anyDouble(), anyDouble()))
+            .thenAnswer(invocation -> new LaplaceNoise().computeConfidenceInterval(
+                    (Long) invocation.getArguments()[0], (Integer) invocation.getArguments()[1], (Long) invocation.getArguments()[2],
+                    (Double) invocation.getArguments()[3], null, (Double) invocation.getArguments()[5]));
+    count = Count.builder()
+            .epsilon(0.1)
+            .noise(noise)
+            .delta(0.5)
+            .maxPartitionsContributed(1)
```

Comment:
can't use default count?

Reply:
For the sake of trying out different count cases, i tried different values. Should i change all the tests to default count?

Reply:
yes, if these different count are not the main thing which you are testing in each test